packages/react-form/src/types.ts

```diff
@@ -28,3 +29,12 @@ export type UseFieldOptions<
 > & {
   mode?: 'value' | 'array'
 }
+
+/**
+ * The return type use useForm with pre-populated generics
+ */
+export type UseFormReturnType<TForm, TFormValidator> = TFormValidator extends
+  | Validator<TForm, unknown>
+  | undefined
+  ? ReturnType<typeof useForm<TForm, TFormValidator>>
+  : never
```

Comment:
I'm not certain this is required, but it cleans up the types on the components. Let me know what you think 🤟

Reply:
ah, this Isn't needed this can just be simplified on the user to  ReactFormExtendedApi<TForm, any>;